asg/src/expression/call.rs

```diff
@@ -108,16 +109,18 @@ impl<'a> FromAst<'a, leo_ast::CallExpression> for CallExpression<'a> {
                             .into());
                         }
                         Some(Type::Array(_, _)) | Some(Type::ArrayWithoutSize(_)) => {
-                            scope.resolve_circuit("array").unwrap()
+                            scope.resolve_circuit(sym::array).unwrap()
                         }
                         Some(Type::Circuit(circuit)) => circuit,
-                        Some(Type::Integer(int_type)) => scope.resolve_circuit(&int_type.to_string()).unwrap(),
-                        Some(type_) => scope.resolve_circuit(&type_.to_string()).unwrap(),
+                        Some(Type::Integer(int_type)) => {
+                            scope.resolve_circuit(Symbol::intern(&int_type.to_string())).unwrap()
```

Comment:
Can you explain how this works? I looked at the intern function and saw it does a symbol lookup. But I did not see symbols of the different int types. Does this build a symbol on the fly based on the string in that case?

Otherwise everything else looked good to me.

Reply:
Yeah in this case it builds a symbol on the fly. Not every core language concept has been added as a predefined symbol yet. That's future work for now.